### ISSUE TITLE
Option `struct_lit_single_line` formats Configurations.md example with either value

### ISSUE THREAD

**Issue by davidalber** [rust-lang-nursery/rustfmt#2458]
Configurations.md has (almost) the following example for the `struct_lit_single_line = true` example ([link](https://github.com/rust-lang-nursery/rustfmt/blob/f2bed291baf8952c78fbd1e74157234a8acdbfd3/Configurations.md#struct_lit_single_line)).
```rust
fn main() {
    let lorem = Lorem { ipsum: dolor, sit: amet };
}
```

However, given a rustfmt.toml containing only `struct_lit_single_line = true` and formatting with `--write-mode=diff` the result is the following.
```diff
 fn main() {⏎
-    let lorem = Lorem { ipsum: dolor, sit: amet };⏎
+    let lorem = Lorem {⏎
+        ipsum: dolor,⏎
+        sit: amet,⏎
+    };⏎
 }⏎
 ⏎
```

This is the same result when rustfmt.toml contains only `struct_lit_single_line = false`.

**Comment by nrc** [comment]
IIRC, what happens might depend on the 'short' heuristics values. So it might be that a few characters shorted makes the example work?

**Comment by davidalber** [comment]
You are right. If I change the example to the following it is left alone when `struct_lit_single_line = true`.
```rust
fn main() {
    let lorem = Lorem { foo: bar, baz: ofo };
}
```

That's the longest it can be (by default, at least), however. The following does get broken up.
```rust
fn main() {
    let lorem = Lorem { foo: bar, baz: quux };
}
```

**Comment by topecongiro** [comment]
Closed via #2469.
